File: 2024/day01-1.ts
function solution(content: string) {
  const split = content
    .split("\n")
    .map((line) => line.split("   ").map((item) => parseInt(item)));
  const left = split.map((item) => item.at(0)!).toSorted((a, b) => a - b);
  const right = split.map((item) => item.at(1)!).toSorted((a, b) => a - b);
  return left
    .map((l, index) => Math.abs(l - right[index]))
    .reduce((acc, cur) => acc + cur, 0);
}

Deno.readTextFile("day01.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);

File: 2024/day01_1.py
# pylint: disable=missing-module-docstring, missing-function-docstring


from typing import Any


def solution(content: list[str]) -> Any:
    split = [line.split() for line in content]
    left = sorted(int(a) for a, *_ in split)
    right = sorted(int(a) for *_, a in split)
    return sum(abs(a - b) for a, b in zip(left, right))


def main() -> None:
    with open("day01.dat", "r", encoding="utf8") as file:
        print(solution(file.readlines()))


if __name__ == "__main__":
    main()
